tt_metal/impl/buffers/buffer.hpp

```diff
@@ -179,7 +179,7 @@ class Buffer {
 
     void set_size(uint64_t size) { size_ = size; }
     // Returns address of buffer in the first bank
-    uint32_t address() const { return static_cast<uint32_t>(address_); }
+    virtual uint32_t address() const { return static_cast<uint32_t>(address_); }
```

Comment:
please make original buffer destructor virtual too.

Reply:
This was done in a previous commit.